### ISSUE TITLE
Table UI view freezes

### ISSUE THREAD

**Issue by PedroCostaMudey** [dbeaver/dbeaver#21242]
### Description

The table UI freezes and I have to restart Dbeaver.
I still can press some buttons and some of other parts other then the table view show changes (very slowly).

### DBeaver Version

Community Edition 23.2.0

### Operating System

Ubuntu 22.04

### Database and driver

mariadb mysql:8.0.20

docker container

### Steps to reproduce

This only happens sometimes when I change to other application and come back with alt tab.

**Comment by Destrolaric** [comment]
Hi, Could you please try to generate a thread dump when it happens and attach it here? Here is the instruction: https://dbeaver.com/docs/wiki/Making-a-thread-dump/
Thank you.

**Comment by bodewig** [comment]
I'm not the original reporter but I'm seeing the same using dbeaver-ce installed via snap. While trying to create a thread dump I noticed pgrep simply doesn't find the PID and running jstack in the process manually results in

```
Unable to open socket file /proc/730091/root/tmp/.java_pid730091: target process 730091 doesn't respond within 10500ms or HotSpot VM not loaded
```

I'll use a version installed manually and report whether this changes things.

**Comment by bodewig** [comment]
I'm not a heavy user of dbeaver myself. Yesterday I used it for a few hours and had freezes every quarter of an hour. After switching to a non-snap version and working with it for a few more hours I have not seen any freezes at all. This doesn't proof there are no freezes anymore, of course.

Maybe other people facing the same problem can try the "nojdk" tar.gz (which is what I used).

BTW whenever dbeaver wants to show a popup I see GTK logging a warning like

```
(DBeaver:759071): Gtk-WARNING **: 12:38:26.872: Negative content width -8 (allocation 5, extents 6x7) while allocating gadget (node separator, owner GtkSeparatorToolItem)
```

**Comment by lubakka** [comment]
I have the same problem. Can you fix it? I'll wait for a fix.

**Comment by DandyEPurwanto** [comment]
i got the same issue, its almost always happened after I minimize the dbeaver

**Comment by jtcours** [comment]
I'm running into a similar issue, usually after minimizing dbeaver and then expanding it again. Ubuntu 22.04.3 LTS running xwayland 22.1.1-1ubuntu0.7 and gnome.

Ubuntu 22.04 uses snap to run dbeaver, which (I think) causes the [standard instructions](https://dbeaver.com/docs/dbeaver/Making-a-thread-dump/) for getting a thread dump to fail and seems to interfere with jstack.

```
$ pgrep dbeaver
$ pgrep dbeaver-ce
$ pgrep '/snap/dbeaver-ce/'
$ ps -e -o pid,args | grep dbeaver
  87081 /snap/dbeaver-ce/268/usr/share/dbeaver-ce/jre/bin/java -XX:+IgnoreUnrecognizedVMOptions -Dosgi.requiredJavaVersion=17 -Dfile.encoding=UTF-8 --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.nio=ALL-UNNAMED --add-opens=java.base/java.nio.charset=ALL-UNNAMED --add-opens=java.base/java.text=ALL-UNNAMED --add-opens=java.base/java.time=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED --add-opens=java.base/jdk.internal.vm=ALL-UNNAMED --add-opens=java.base/jdk.internal.misc=ALL-UNNAMED --add-opens=java.base/sun.nio.ch=ALL-UNNAMED --add-opens=java.base/sun.security.ssl=ALL-UNNAMED --add-opens=java.base/sun.security.action=ALL-UNNAMED --add-opens=java.base/sun.security.util=ALL-UNNAMED --add-opens=java.security.jgss/sun.security.jgss=ALL-UNNAMED --add-opens=java.security.jgss/sun.security.krb5=ALL-UNNAMED --add-opens=java.desktop/java.awt=ALL-UNNAMED --add-opens=java.desktop/java.awt.peer=ALL-UNNAMED --add-opens=java.sql/java.sql=ALL-UNNAMED -Xms64m -Xmx1024m -Ddbeaver.distribution.type=deb -jar /snap/dbeaver-ce/268/usr/share/dbeaver-ce/plugins/org.eclipse.equinox.launcher_1.6.500.v20230717-2134.jar -os linux -ws gtk -arch x86_64 -showsplash -launcher /snap/dbeaver-ce/268/usr/share/dbeaver-ce/dbeaver -name Dbeaver --launcher.library /snap/dbeaver-ce/268/usr/share/dbeaver-ce/plugins/org.eclipse.equinox.launcher.gtk.linux.x86_64_1.2.700.v20221108-1024/eclipse_11801.so -startup /snap/dbeaver-ce/268/usr/share/dbeaver-ce/plugins/org.eclipse.equinox.launcher_1.6.500.v20230717-2134.jar --launcher.overrideVmargs -exitdata e8038 -vm /snap/dbeaver-ce/268/usr/share/dbeaver-ce/jre/bin/java -vmargs -XX:+IgnoreUnrecognizedVMOptions -Dosgi.requiredJavaVersion=17 -Dfile.encoding=UTF-8 --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.nio=ALL-UNNAMED --add-opens=java.base/java.nio.charset=ALL-UNNAMED --add-opens=java.base/java.text=ALL-UNNAMED --add-opens=java.base/java.time=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED --add-opens=java.base/jdk.internal.vm=ALL-UNNAMED --add-opens=java.base/jdk.internal.misc=ALL-UNNAMED --add-opens=java.base/sun.nio.ch=ALL-UNNAMED --add-opens=java.base/sun.security.ssl=ALL-UNNAMED --add-opens=java.base/sun.security.action=ALL-UNNAMED --add-opens=java.base/sun.security.util=ALL-UNNAMED --add-opens=java.security.jgss/sun.security.jgss=ALL-UNNAMED --add-opens=java.security.jgss/sun.security.krb5=ALL-UNNAMED --add-opens=java.desktop/java.awt=ALL-UNNAMED --add-opens=java.desktop/java.awt.peer=ALL-UNNAMED --add-opens=java.sql/java.sql=ALL-UNNAMED -Xms64m -Xmx1024m -Ddbeaver.distribution.type=deb -jar /snap/dbeaver-ce/268/usr/share/dbeaver-ce/plugins/org.eclipse.equinox.launcher_1.6.500.v20230717-2134.jar
  89179 grep --color=auto dbeaver
$ which jstack
$ find /snap/dbeaver-ce/268/ -name jstack -print
/snap/dbeaver-ce/268/usr/share/dbeaver-ce/jre/bin/jstack
$ /snap/dbeaver-ce/268/usr/share/dbeaver-ce/jre/bin/jstack 87081
87081: Unable to open socket file /proc/87081/root/tmp/.java_pid87081: target process 87081 doesn't respond within 10500ms or HotSpot VM not loaded
$ 
```

Edit: here's some more information. When I minimize dbeaver with a SQL script open, I get these messages in /var/log/syslog:
```
Nov 16 11:22:46 jcours-framework kernel: [190457.193287] audit: type=1400 audit(1700162566.536:1244): apparmor="DENIED" operation="open" class="file" profile="snap.dbeaver-ce.dbeaver-ce" name="/etc/pulse/client.conf" pid=87081 comm="java" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Nov 16 11:22:46 jcours-framework kernel: [190457.193774] audit: type=1400 audit(1700162566.536:1245): apparmor="DENIED" operation="connect" class="file" profile="snap.dbeaver-ce.dbeaver-ce" name="/run/user/1000/pulse/native" pid=87081 comm="java" requested_mask="wr" denied_mask="wr" fsuid=1000 ouid=1000
Nov 16 11:22:53 jcours-framework gnome-shell[2375]: Window manager warning: Ping serial 190459076 was reused for window W794, previous use was for window 0xa001c9.
Nov 16 11:23:16 jcours-framework gnome-shell[2375]: Window manager warning: Ping serial 190482084 was reused for window 0xa001c9, previous use was for window W794.
Nov 16 11:23:20 jcours-framework kernel: [190491.162320] audit: type=1400 audit(1700162600.504:1246): apparmor="DENIED" operation="open" class="file" profile="snap.dbeaver-ce.dbeaver-ce" name="/etc/pulse/client.conf" pid=87081 comm="java" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Nov 16 11:23:20 jcours-framework kernel: [190491.162536] audit: type=1400 audit(1700162600.504:1247): apparmor="DENIED" operation="connect" class="file" profile="snap.dbeaver-ce.dbeaver-ce" name="/run/user/1000/pulse/native" pid=87081 comm="java" requested_mask="wr" denied_mask="wr" fsuid=1000 ouid=1000
```
Expanding it after a while produces these and a hung SQL editor window:
```
Nov 16 11:28:39 jcours-framework kernel: [190810.225785] audit: type=1400 audit(1700162919.560:1282): apparmor="DENIED" operation="open" class="file" profile="snap.dbeaver-ce.dbeaver-ce" name="/etc/pulse/client.conf" pid=87081 comm="java" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Nov 16 11:28:39 jcours-framework kernel: [190810.225963] audit: type=1400 audit(1700162919.560:1283): apparmor="DENIED" operation="connect" class="file" profile="snap.dbeaver-ce.dbeaver-ce" name="/run/user/1000/pulse/native" pid=87081 comm="java" requested_mask="wr" denied_mask="wr" fsuid=1000 ouid=1000
```
Clicking within the SQL script or the Databases tab produces no response. The menus (File, Help, etc.) at the top still pop up when I click on them, and I'm able to select a different databsae on the same connection through the database selection dropdown.

Mousing over the database connections sometimes produces the popup with name, url, and user, but it's inconsistent and sends these messages to /var/log/syslog:
```
Nov 16 11:37:00 jcours-framework kernel: [191311.227927] audit: type=1400 audit(1700163420.553:1400): apparmor="DENIED" operation="open" class="file" profile="snap.dbeaver-ce.dbeaver-ce" name="/etc/pulse/client.conf" pid=87081 comm="java" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Nov 16 11:37:00 jcours-framework kernel: [191311.228152] audit: type=1400 audit(1700163420.553:1401): apparmor="DENIED" operation="connect" class="file" profile="snap.dbeaver-ce.dbeaver-ce" name="/run/user/1000/pulse/native" pid=87081 comm="java" requested_mask="wr" denied_mask="wr" fsuid=1000 ouid=1000
```
Clicking in the SQL scirpt window and trying to type produces no response.

**Comment by DandyEPurwanto** [comment]
any update on this issue ?

**Comment by petrisorciprian-vitals** [comment]
Confirming, also experiencing this issue with snap.
Version installed via apt on Debian using the .deb package has no such problems.

**Comment by lubakka** [comment]
The problem is still not solved. I'm still waiting on fix.

**Comment by martinyonatann** [comment]
i have same problem on ubuntu

**Comment by petrisorciprian-vitals** [comment]
@mayerro  
Could you kindly provide some information as to why this was closed? 
Was it fixed/will it be fixed?
If so, any ETA available ?

Thank you for your work!

**Comment by batosti** [comment]
I got the same issue here, have any workarround that we can do?

**Comment by Destrolaric** [comment]
@petrisorciprian-vitals This should have been fixed in this release. Are you using snap release?

**Comment by petrisorciprian-vitals** [comment]
@Destrolaric 
I tried the snap release a couple of days ago, version 23.3.2 of dbeaver.

After some time, the UI would freeze up, and I'd have to restart dbeaver in order to get it to work again.
I noticed it especially does this when I minimze and bring the window up again.
This happened consistently throughout the day. I experienced the same behaviour on the prior release (23.3.1), also instaled via snap.

The .deb has no such issues, so I reverted to that in the meantime.

**Comment by Dzyamon** [comment]
Hi,
For me Dbeaver (Version 23.3.2.202401072024) main window freezes but upper nav menu is available and when exiting asks about saving as usual.
I noticed this freeze bug when switching windows with "Alt+Tab" combination on Ubuntu 22.04.3 LTS Wayland (Gnome v42.9).
When clicking separate windows as usual no freeze.

**Comment by Destrolaric** [comment]
Hi everyone, during the freeze, could anyone try to make a thread dump
`jstack -e $(ps aux | grep java | grep dbeaver | awk '{print $2}')  > ./dbeaver-thread-dump.txt`

**Comment by lubakka** [comment]
> Hi everyone, during the freeze, could anyone try to make a thread dump `jstack -e $(ps aux | grep java | grep dbeaver | awk '{print $2}') > ./dbeaver-thread-dump.txt`

This is my dump. I hope it helps you

[dbeaver-thread-dump.txt](https://github.com/dbeaver/dbeaver/files/14048620/dbeaver-thread-dump.txt)

**Comment by S3sc0** [comment]
> @Destrolaric I tried the snap release a couple of days ago, version 23.3.2 of dbeaver.
> 
> After some time, the UI would freeze up, and I'd have to restart dbeaver in order to get it to work again. I noticed it especially does this when I minimze and bring the window up again. This happened consistently throughout the day. I experienced the same behaviour on the prior release (23.3.1), also instaled via snap.
> 
> The .deb has no such issues, so I reverted to that in the meantime.

I'd like to confirm that dbeaver installed with .deb package is solving the issue in contrary to the version installed with snap

**Comment by ShadelessFox** [comment]
Hello @lubakka,

The thread dump you've provided doesn't contain anything useful. Are you sure the dump was made after DBeaver started "freezing"?

**Comment by Destrolaric** [comment]
@lubakka Hi. We still need additional information. Can you please try making the thread dump again?

**Comment by lubakka** [comment]
Hallo @Destrolaric 

Now I have another problem with DBeaver and I had to go back to 23.3.0.202312122044 version because when I open any table with many records it crashes. So first I have to wait for the new version from which it is currently so that I can install it on the new version and see if the freeze is there or not.

With each new version of DBeaver I have different problems.

**Comment by ShadelessFox** [comment]
@lubakka We can't fix something we can't reproduce. Please provide at least some details regarding your issues.

**Comment by lubakka** [comment]
@ShadelessFox 
Here is the error I am getting now with the new version "Version 23.3.4.202402060628" And I use a limit of 200 rows

![Bildschirmfoto vom 2024-02-18 16-15-00](https://github.com/dbeaver/dbeaver/assets/5030587/aa40229d-e5f5-4c3a-a181-dbaca1c90fcd)
![Bildschirmfoto vom 2024-02-18 16-12-22](https://github.com/dbeaver/dbeaver/assets/5030587/578258a6-6ba0-48c9-a193-06210ccfd721)
![Bildschirmfoto vom 2024-02-18 16-12-14](https://github.com/dbeaver/dbeaver/assets/5030587/070877c3-74e1-487c-9ab1-7fd43ebaaac9)
![Bildschirmfoto vom 2024-02-18 16-17-36](https://github.com/dbeaver/dbeaver/assets/5030587/f32ca199-9b22-49dc-ac33-eb193f1199b5)



-------------------------------------------------------------------------------------------------
java.lang.OutOfMemoryError: Java heap space
	at org.eclipse.swt.graphics.Image.getImageDataAtCurrentZoom(Image.java:1053)
	at org.eclipse.swt.graphics.Image.getImageData(Image.java:1124)
	at org.eclipse.swt.graphics.Image.getImageData(Image.java:1013)
	at org.eclipse.swt.graphics.GC.drawImage(GC.java:860)
	at org.eclipse.swt.graphics.GC.drawImageInPixels(GC.java:810)
	at org.eclipse.swt.graphics.GC.drawImage(GC.java:807)
	at org.jkiss.dbeaver.ui.controls.ProgressLoaderVisualizer.lambda$0(ProgressLoaderVisualizer.java:167)
	at org.jkiss.dbeaver.ui.controls.ProgressLoaderVisualizer$$Lambda$1096/0x00000001009c4af8.paintControl(Unknown Source)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:234)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5854)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1529)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1555)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1538)
	at org.eclipse.swt.widgets.Control.gtk_draw(Control.java:3927)
	at org.eclipse.swt.widgets.Scrollable.gtk_draw(Scrollable.java:365)
	at org.eclipse.swt.widgets.Composite.gtk_draw(Composite.java:506)
	at org.eclipse.swt.widgets.Canvas.gtk_draw(Canvas.java:174)
	at org.eclipse.swt.widgets.Widget.windowProc(Widget.java:2482)
	at org.eclipse.swt.widgets.Control.windowProc(Control.java:6883)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:6162)
	at org.eclipse.swt.internal.gtk3.GTK3.gtk_main_do_event(Native Method)
	at org.eclipse.swt.widgets.Display.eventProc(Display.java:1597)
	at org.eclipse.swt.internal.gtk3.GTK3.gtk_main_iteration_do(Native Method)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4513)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:342)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:648)
	at org.eclipse.ui.internal.Workbench$$Lambda$220/0x0000000100377490.run(Unknown Source)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:342)


--------------------
eclipse.buildId=unknown
java.version=17.0.6
java.vendor=Eclipse Adoptium
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=de_DE
Framework arguments:  -eclipse.keyring /home/lb/.local/share/DBeaverData/secure/secure_storage
Command-line arguments:  -os linux -ws gtk -arch x86_64

**Comment by ShadelessFox** [comment]
Please record a video showing the issue. It's not clear how this was achieved.

**Comment by lubakka** [comment]
@ShadelessFox 
Sorry, but I'm using it in German. And a message comes up about not enough space.

[Bildschirmaufzeichnung vom 18.02.2024, 16:26:18.webm](https://github.com/dbeaver/dbeaver/assets/5030587/5d06aa71-c595-4ddc-b23f-5eeb8336a693)

**Comment by ShadelessFox** [comment]
I see, thank you. Try enabling this option to see if it helps:

![image](https://github.com/dbeaver/dbeaver/assets/35821147/63f284fe-6ecd-4720-839d-1f8594ab9d6b)

**Comment by lubakka** [comment]
@ShadelessFox 
That doesn't help, same error again.

**Comment by ShadelessFox** [comment]
Could you provide the DDL for that table and the driver/server version?

**Comment by lubakka** [comment]
@ShadelessFox 

I have MariaDB with version: 10.3.39-MariaDB-1:10.3.39+maria~ubu2004


CREATE TABLE `changes` (
  `_changesID` int(11) NOT NULL AUTO_INCREMENT,
  `DataID` int(11) NOT NULL DEFAULT 0,
  `DataTabelle` varchar(255) DEFAULT '',
  `Aktion` varchar(255) DEFAULT '',
  `GrundDataID` int(11) NOT NULL DEFAULT 0,
  `GrundDataTabelle` varchar(255) NOT NULL DEFAULT '',
  `Aenderung` text DEFAULT NULL,
  `Nutzername` varchar(255) DEFAULT '',
  `Zeitpunkt` datetime DEFAULT '0000-00-00 00:00:00',
  `_updated` timestamp NULL DEFAULT current_timestamp() ON UPDATE current_timestamp(),
  `_updatedby` varchar(32) DEFAULT '',
  `_created` datetime DEFAULT '0000-00-00 00:00:00',
  `_createdby` varchar(32) DEFAULT '',
  PRIMARY KEY (`_changesID`),
  KEY `idx2` (`DataTabelle`) USING BTREE,
  KEY `idx4` (`GrundDataTabelle`) USING BTREE,
  KEY `idx5` (`Aktion`) USING BTREE,
  KEY `idx6` (`Zeitpunkt`) USING BTREE,
  KEY `idx1` (`DataID`) USING BTREE,
  KEY `idx3` (`GrundDataID`) USING BTREE,
  KEY `idx7` (`_created`) USING BTREE,
  KEY `idx8` (`_createdby`) USING BTREE
) ENGINE=InnoDB AUTO_INCREMENT=10529992 DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci ROW_FORMAT=FIXED;



![Bildschirmfoto vom 2024-02-18 17-11-30](https://github.com/dbeaver/dbeaver/assets/5030587/33cb47ad-fc52-4b2a-863c-0205721c31a4)

**Comment by ShadelessFox** [comment]
Thank you. I don't see anything suspicious. The issue might be with the driver.

Let's try a different approach. Try executing the following query in the SQL editor:
```sql
SELECT * FROM integris.changes LIMIT 200;
```

Adjust the limit to see when it starts consuming a lot of memory.

**Comment by lubakka** [comment]
@ShadelessFox 

Yes it works when I run the query it works it shows 200 results
And it shows an error again when the limit is 20000000

```sql
SELECT * FROM integris.changes LIMIT 20000000; 
```

> Thank you. I don't see anything suspicious. The issue might be with the driver.
> 
> Let's try a different approach. Try executing the following query in the SQL editor:
> 
> ```sql
> SELECT * FROM integris.changes LIMIT 200;
> ```
> 
> Adjust the limit to see when it starts consuming a lot of memory.

**Comment by lubakka** [comment]
@ShadelessFox 
I installed now the new version 23.3.5 and again I have a problem

![Bildschirmfoto vom 2024-02-19 09-31-01](https://github.com/dbeaver/dbeaver/assets/5030587/4525312c-e180-4457-817c-50d7aebb352d)
![Bildschirmfoto vom 2024-02-19 09-30-55](https://github.com/dbeaver/dbeaver/assets/5030587/3412513a-284e-4eb9-a829-6685b5516fe2)

-----------------------------------------------------------
java.lang.OutOfMemoryError: Java heap space
	at org.eclipse.swt.widgets.Scrollable.vScrollBarSize(Scrollable.java:592)
	at org.eclipse.swt.widgets.Scrollable.vScrollBarWidth(Scrollable.java:581)
	at org.eclipse.swt.widgets.Scrollable.computeTrimInPixels(Scrollable.java:131)
	at org.eclipse.swt.widgets.Scrollable.computeTrim(Scrollable.java:120)
	at org.eclipse.swt.widgets.Composite.computeSizeInPixels(Composite.java:281)
	at org.eclipse.swt.widgets.Control.computeSize(Control.java:837)
	at org.eclipse.swt.widgets.Control.computeSize(Control.java:693)
	at org.eclipse.e4.ui.workbench.renderers.swt.TrimBarLayout.computeSize(TrimBarLayout.java:178)
	at org.eclipse.e4.ui.workbench.renderers.swt.TrimBarLayout$TrimLine.addControl(TrimBarLayout.java:52)
	at org.eclipse.e4.ui.workbench.renderers.swt.TrimBarLayout.computeSize(TrimBarLayout.java:141)
	at org.eclipse.swt.widgets.Composite.computeSizeInPixels(Composite.java:269)
	at org.eclipse.swt.widgets.Control.computeSize(Control.java:837)
	at org.eclipse.e4.ui.workbench.renderers.swt.TrimmedPartLayout.layout(TrimmedPartLayout.java:132)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1877)
	at org.eclipse.swt.widgets.Composite.setLayoutDeferred(Composite.java:1720)
	at org.eclipse.swt.widgets.Display.runDeferredLayouts(Display.java:5088)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4504)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:342)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:648)
	at org.eclipse.ui.internal.Workbench$$Lambda$233/0x00000001003ad4c0.run(Unknown Source)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:342)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:555)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:290)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)


--------------------------------------------------------------------------------
eclipse.buildId=unknown
java.version=17.0.6
java.vendor=Eclipse Adoptium
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=de_DE
Framework arguments:  -eclipse.keyring /home/lb/.local/share/DBeaverData/secure/secure_storage
Command-line arguments:  -os linux -ws gtk -arch x86_64

**Comment by petrisorciprian-vitals** [comment]
I'll attempt to extract a trace as well, reaching back when I have it.

**Comment by petrisorciprian-vitals** [comment]
@ShadelessFox , using the snap release, I'm getting the following error when attempting a thread dump:
![image](https://github.com/dbeaver/dbeaver/assets/136788902/56f68cb5-ad36-4ff2-bdf8-b40139abeaf8)
I used the jstack bundled with the snap release, as you can see.

Visually, nothing under the top menu bar of dbeaver works (File, Edit, Navigate, etc all work and I can bring up new windows). Resizing the window leads to black borders.

The only solution is to close dbeaver and start it again; it seems like a crash/canvas painting bug based on the above.

Any pointers on how to get a thread dump/circumvent this issue?

**Comment by ShadelessFox** [comment]
Sorry, I don't know how to create thread dumps inside snap packages. It would be simpler to use a non-snap distribution.

**Comment by petrisorciprian-vitals** [comment]
The problem does not reproduce using non-snap releases, though.

**Comment by ShadelessFox** [comment]
Interesting, thank you for pointing that out. I'm going to discuss this issue with my colleagues.

**Comment by crish1eev1** [comment]
I've tried to make a thread-dump while dbeaver was freezing. It happened after minimizing it for a while and opening it again. 

As described by few people already, the top bar (File, Edit, Navigate, ..., Help) was the only part not frozen. 

[dbeaver-thread-dump.txt](https://github.com/dbeaver/dbeaver/files/14449495/dbeaver-thread-dump.txt)

**Comment by rgriffogoes** [comment]
I'm using it in Windows, currently on version `Version 23.3.4.202402060628`  but since a few last minor releases I'm facing freezes rather frequent. Couldn't pin point exact operation that triggers it, but the result is the usual "_Not Responding_" application, and I have to kill it.

My frequently used DB connections these days have been Postgres and Snowflake.

Here's a thread dump of while the it was "not responding"

[dbeaver-thread-dump.txt](https://github.com/dbeaver/dbeaver/files/14471991/dbeaver-thread-dump.txt)

**Comment by LonwoLonwo** [comment]
Hello @rgriffogoes 

This issue is more about snap-problem. Please create a separate issue for your case on WIndows.

**Comment by Oleksandr-thescimus** [comment]
I've encountered an issue with DBeaver freezing. Initially, it froze only occasionally, but recently it started freezing consistently within 2-5 minutes after opening.

My environment:
Operating System: Ubuntu 22.04.4
DBeaver Version: 24.0.0  (the issue started in an older version, and I have tried updating to fix it, but it didn't help)

**Solution**:
To address this issue, I found that running the application via the command line provided a workaround.
Just open a terminal and run: `dbeaver` or `dbeaver-ce`

**Comment by brenoatgetmomo** [comment]
> I've encountered an issue with DBeaver freezing. Initially, it froze only occasionally, but recently it started freezing consistently within 2-5 minutes after opening.
> 
> My environment: Operating System: Ubuntu 22.04.4 DBeaver Version: 24.0.0 (the issue started in an older version, and I have tried updating to fix it, but it didn't help)
> 
> **Solution**: To address this issue, I found that running the application via the command line provided a workaround. Just open a terminal and run: `dbeaver` or `dbeaver-ce`

I'm having the same problem same environment.

**Comment by brenoatgetmomo** [comment]
> I've encountered an issue with DBeaver freezing. Initially, it froze only occasionally, but recently it started freezing consistently within 2-5 minutes after opening.
> 
> My environment: Operating System: Ubuntu 22.04.4 DBeaver Version: 24.0.0 (the issue started in an older version, and I have tried updating to fix it, but it didn't help)
> 
> **Solution**: To address this issue, I found that running the application via the command line provided a workaround. Just open a terminal and run: `dbeaver` or `dbeaver-ce`

I'm having the same problem same environment.

terminal doesn work and i have this error:
```
(DBeaver:181460): Gtk-WARNING **: 11:25:04.162: Negative content width -8 (allocation 5, extents 6x7) while allocating gadget (node separator, owner GtkSeparatorToolItem)
```

**Comment by jimddavis** [comment]
I had a similar problem on DBeaver for Windows  with script screen freezing when truncating multiple remote tables(10-20).  The truncate completed, It froze refreshing the UI after truncation.  

For completeness, I was clearing the Drupal cache tables.  I filtered the full table display using cache*, did a Select All, then Tools-Truncate Tables.

 Found a post somewhere indicating that increasing the memory allocated to DBeaver may help.  It worked for me.

Open dbeaver.ini in the installation folder.  The memory setting is at the bottom, -Xmx1024m.  Per recommendation I changed it to -Xmx4096m, 4GB, and the problem went away.  

I am using a the following laptop
Device name	LenovaIPG3
Processor	AMD Ryzen 7 5800H with Radeon Graphics            3.20 GHz
Installed RAM	16.0 GB (13.9 GB usable)
Windows 11 23H2 Build 22631.3593

**Comment by shubhambarudwale** [comment]
Facing the same issue.

**Comment by brenoatgetmomo** [comment]
Installing by downloading from the website, instead of using snap solved to me.
The downside its not updating it automatically and every two days receiving the message to update it.

**Comment by monterhealth** [comment]
Facing the same very annoying issue.

DBeaver freezes all the time when I navigate to another window and click back to DBeaver. Restarting DBeaver is then required. Is there any way to solve this? This is not workable.

Operating System: Ubuntu 22.04.4 LTS
DBeaver version: 24.2.0.202409011551

**Comment by ShadelessFox** [comment]
Hi @monterhealth,

Unfortunately, such a complaint won't give us any insight into what actually causes your issue. Instead, during a freeze, please make a thread dump, as well as log files.

You can find instructions here: 
- https://github.com/dbeaver/dbeaver/wiki/Making-a-thread-dump
- https://github.com/dbeaver/dbeaver/wiki/Log-files#dbeaver-log-files

**Comment by Tuyuyi99** [comment]
Hello,

I think I just resolved this bug in my computer.
It is someway related with Wayland. I changed it to Xorg And it works like a charm. Have you tried that?

To change from wayland to Xorg you have to edit the next files:

/etc/gdm3/custom.conf

In this file, uncomment the line with "WaylandEnable=false"

~/.bashrc or ~/.zshrc

Add the next line at the end of the file:
export QT_QPA_PLATFORM=xcb

And restart the computer.

I hope it helps :)

**Comment by khalt00** [comment]
Hi, @Tuyuyi99 I haven't try your solution yet but will this affect other apps ?
I think the fix should come from dbeaver, not our machine

**Comment by Tuyuyi99** [comment]
> Hi, @Tuyuyi99 I haven't try your solution yet but will this affect other apps ? I think the fix should come from dbeaver, not our machine

Yes, it will affect other apps too, but it's the only solution I found for this problem :(

**Comment by vlasvlasvlas** [comment]
same here

**Comment by bmelman** [comment]
DBeaver has been freezing on multimonitor systems running Ubuntu (at least 18.x till 24.04) all the time, usually after switching focus to a different screen or workspace.

**Comment by faireks** [comment]
Same problem

**Comment by ShadelessFox** [comment]
Please don't post unconsolidated comments like "+1" or "same issue"; they are pointless and don't shine any light on the issue. Instead, provide at least some basic information such as OS name and DBeaver version. This alone will help us greatly.

**Comment by faireks** [comment]
I deleted my comment from guthub.

Version DBeaver: Version 24.3.5.202502162012 (and previous).
OS: Ubuntu 22.04.1
Two monitors.

**Comment by Arthur-quiteja** [comment]
> > Hi, [@Tuyuyi99](https://github.com/Tuyuyi99) I haven't try your solution yet but will this affect other apps ? I think the fix should come from dbeaver, not our machine
> 
> Yes, it will affect other apps too, but it's the only solution I found for this problem :(

did you have a example of what kind of issues this fix can cause in ohter apps?

**Comment by guybrand** [comment]
Hi.

I'm having the same issue, Ubuntu 24.04.2 LTS, multiple displays, Dbeaver Version 25.0.5.202505181758.
I cannot go back to X cause I it causes other issues, especially with screen sharing and preformance.

The freeze happens to me various times a day, so I can easily reconstruct and run any needed command to debug, and assist in resolving this, just LMK

**Comment by jimddavis** [comment]
What solved my problem was allocating more memory to DBeaver processes.  This is done in the .ini file.  Below was copied from Googles AI response, but is exactly how I did it.

Here's how to do it:
Locate the dbeaver.ini file: This file is typically found in the DBeaver installation directory or in your user's application data folder (e.g., C:\Users\<User>\AppData\Local\DBeaver). 
Open the dbeaver.ini file: You can open it with a text editor like Notepad. 
Find the -Xmx parameter: Look for a line that starts with -Xmx (e.g., -Xmx1024m). This specifies the maximum memory in megabytes (MB) that DBeaver can use. 
Change the value: Increase the number after -Xmx to your desired memory limit (e.g., -Xmx4096m for 4GB). You can also set the initial memory with -Xms (e.g., -Xms2048m for 2GB). 
Save the file: Save the modified dbeaver.ini file.
Restart DBeaver: You'll need to restart DBeaver for the changes to take effect.

**Comment by Tuyuyi99** [comment]
> > > Hi, [@Tuyuyi99](https://github.com/Tuyuyi99) I haven't try your solution yet but will this affect other apps ? I think the fix should come from dbeaver, not our machine
> > 
> > 
> > Yes, it will affect other apps too, but it's the only solution I found for this problem :(
> 
> did you have a example of what kind of issues this fix can cause in ohter apps?

Hello. I personally didn't experimented any problem from the apps that I use for my work, so I don't know what problems can cause. As far as I know, it works like a charm for me.
